### PR TITLE
Add updatepayment handler

### PR DESCRIPTION
**Pull request by mdayican**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/BPS-766

### Change description ###

add handler to handle pay-hub update.
#62   previous stuck in the pipeline. 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageHandlerTest.java [hunk 1]
```diff
@@ -60,4 +68,58 @@ public void should_call_payhub_api_for_successful_payment_message() {
...
+        when(s2sTokenGenerator.generate()).thenReturn("test-service");
+
+        when(payHubClient.updateCaseReference(
+            eq("test-service"), eq("exp-21321"), any(CaseReferenceRequest.class))
+        ).thenThrow(new PayHubClientException(new HttpClientErrorException(HttpStatus.BAD_REQUEST, "BAD_REQUEST")));
+
+        // when
+        assertThatThrownBy(
+            () -> messageHandler.updatePaymentCaseReference(message))
+            .isInstanceOf(PayHubClientException.class);
```
Comment (aliveni) [inline src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageHandlerTest.java]: This test is not needed in my opinion

Comment (doncem) [reply src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageHandlerTest.java]: was about to say "on the other hand" similarly. yes, there's no edge case at the moment to verify. perhaps somewhere higher in the food chain when we need to verify exception handling
